bundles/org.openhab.binding.gardena/src/main/java/org/openhab/binding/gardena/internal/GardenaSmartImpl.java

```diff
@@ -121,14 +125,22 @@ public GardenaSmartImpl(String id, GardenaConfig config, GardenaSmartEventListen
 
             // initially load access token
             verifyToken();
-            locationsResponse = loadLocations();
+            try {
+                locationsResponse = loadLocations();
+            } catch (GardenaException ex) {
+                locationsResponse = null;
+                throw ex;
+            }
```

Comment:
Simplification possible:
```suggestion
            LocationsResponse locationsResponse = loadLocations();
            this.locationsResponse = locationsResponse;
```

and then delete line 136 also (cannot comment across deleted lines). Or perhaps you could simply keep it as is, but without the try/catch. This is the constructor, so we know `locationsResponse` is currently null, thus no need to explicitly set this. The exception will even be caught in line 154, then `dispose()` is called and everything is cleared.

Reply:
Yeah. The problem is that the compiler cannot foresee an exception occurring in the constructor, so it gives spurious null annotation warnings..